src/main/java/io/github/bobocodebreskul/context/scan/utils/ScanUtilsImpl.java

```diff
@@ -60,6 +61,30 @@ public Set<Class<?>> searchClassesByFilter(String packagePath, Predicate<Class<?
   }
 
 
+  @Override
+  public String[] readBasePackages(Class<?> annotatedClass) {
+    BringComponentScan bringComponentScan = annotatedClass.getAnnotation(BringComponentScan.class);
+    if (bringComponentScan != null) {
```

Comment:
You can use method `isAnnotationPresentForClass` here from ReflectionUtils utility class